authorize_request_handler_oidc_request_test.go

```diff
@@ -166,10 +165,11 @@ func TestAuthorizeRequestParametersFromOpenIDConnectRequest(t *testing.T) {
 			expectForm:      url.Values{"scope": {"openid"}},
 		},
 		{
-			d:          "should pass and set request parameters properly",
-			form:       url.Values{"scope": {"openid"}, "request": {validRequestObject}},
-			client:     &DefaultOpenIDConnectClient{JSONWebKeys: jwks, RequestObjectSigningAlgorithm: "RS256"},
-			expectForm: url.Values{"scope": {"foo openid"}, "request": {validRequestObject}, "foo": {"bar"}, "baz": {"baz"}},
+			d:      "should pass and set request parameters properly",
+			form:   url.Values{"scope": {"openid"}, "response_type": {"code"}, "response_mode": {"none"}, "request": {validRequestObject}},
+			client: &DefaultOpenIDConnectClient{JSONWebKeys: jwks, RequestObjectSigningAlgorithm: "RS256"},
+			// The values from form are overwritten by the request object.
+			expectForm: url.Values{"response_type": {"token"}, "response_mode": {"post_form"}, "scope": {"foo openid"}, "request": {validRequestObject}, "foo": {"bar"}, "baz": {"baz"}},
```

Comment:
Any particular reason why this file contains `post_form` and not correct `form_post`? What is this testing? Why it is not failing?